core/src/test/java/com/spotify/metrics/core/InstrumentedExecutorServiceTest.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2018 Spotify AB.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * This class was initially copied and modified from the codahale metrics project.
+ *
+ * For the appropriate license (there was no header) see LICENSE.codahale.txt
+ *
+ * It was copied from the following tree:
+ * https://github.com/dropwizard/metrics/tree/6d1fff844b7fc8855b81bb42b7125bd84f3a3e7a
+ */
+
+package com.spotify.metrics.core;
+
+import static org.junit.Assert.assertEquals;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.Timer;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import org.junit.After;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class InstrumentedExecutorServiceTest {
```

Comment:
Please add a timeout rule to the entire test since we do block a fair bit:
https://github.com/junit-team/junit4/wiki/rules#timeout-rule

Reply:
Fixed in bf3d27c, I took the liberty of bumping JUnit's minor to avoid using Timeout's constructor which was deprecated